### DISCUSSION TITLE
week page dynamic variable, This Week compatibility.

### DISCUSSION THREAD

**Discussion by IvanSoregashi** [YU000jp/logseq-plugin-show-weekday-and-week-number#121] [General]
Hello @YU000jp , first of all, thank you for this extension, looks great.

these are couple of small feature requests:
1. There are are slash command for referring to the /current week, but is it possible to make it a dynamic variable to be used in templates? like <%current week%>
You already have made link week page right after page title, but I want it to be part of the page for various purposes.

2. Currently 'This week' section in weekly page is built based on 'Preferred date format'(page-title-format ?) from the settings. It would be great to have an option to uncouple it, and display weekdays in file-name-format for compatibility purposes, like that:
![image](https://github.com/YU000jp/logseq-plugin-show-weekday-and-week-number/assets/118012131/279fe1a9-8ce8-44e9-ab30-5d7dd47cba46)

3. There is a small bug. If 'This week' section is set to the bottom, Weekly template's first block will be duplicated after the template.
![image](https://github.com/YU000jp/logseq-plugin-show-weekday-and-week-number/assets/118012131/a144fc3d-f536-4b13-a602-03b7686f373f)
It would have been nice if first block metadata was attached to the tags metadata block.
It also will be nice to have a bit more control over the tags metadata block.

**Comment by YU000jp** [comment]
@IvanSoregashi 

1. Currently, Logseq plugin API does not provide additional functionality for dynamic variable. It is difficult to specify dynamic variables from plugins.
You might be able to achieve something similar using [Full House plugin](https://github.com/stdword/logseq13-full-house-plugin). I think it fits the intent of the plug-in, so please consult with the creator.

2. The link must be valid as a journal. This may be possible using Markdown notation. `[file name]([[page title]])`  With this method, there is a possibility that it can be achieved. Is there any problem with this method? If there are no problems, it can be installed.

3. That's a bug. Add it to the list of upcoming bug fixes. Thanks.

**Reply by IvanSoregashi** [reply]
2. It works in logseq just fine. But not in other apps. Lock in is to be avoided. :)
Thank you.

**Reply by YU000jp** [reply]
> 2. It works in logseq just fine. But not in other apps. Lock in is to be avoided. :)

In that case, one or two of my plugins may be helpful.

Select a common date format on Logseq.  (In that case, the old format will be disabled, so activate it with `Old date format` plugin)
https://github.com/YU000jp/logseq-plugin-legacy-date-format

Apart from the file date format, if you want to change the displayed date format, there is a way to use Flexible date format plugin.
https://github.com/YU000jp/logseq-plugin-flex-date-format

**Comment by oraluben** [comment]
> There are are slash command for referring to the /current week, but is it possible to make it a dynamic variable to be used in templates? like <%current week%>

Is it possible to use the title of the current page in the template? After the page's created it should be what you need.

**Reply by YU000jp** [reply]
I don't understand exactly what you want to do, but it could probably be achieved using "full house" plugin.

https://github.com/stdword/logseq13-full-house-plugin/discussions/31

**Comment by YU000jp** [comment]
> > There are are slash command for referring to the /current week, but is it possible to make it a dynamic variable to be used in templates? like <%current week%>
> 
> Is it possible to use the title of the current page in the template? After the page's created it should be what you need.

Output the current page name using 'full house' plugin.
https://stdword.github.io/logseq13-full-house-plugin/#/reference__tags

**Comment by cheerfulstoic** [comment]
👋 I'm trying to use the full house plugin, but I'm not sure how to generate the week number.  Do you have some example code (JS, I guess) which does that?

**Reply by YU000jp** [reply]
It's here. https://github.com/stdword/logseq13-full-house-plugin/discussions/31
